infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationSession.java

```diff
@@ -0,0 +1,69 @@
+package org.corfudb.infrastructure.logreplication.infrastructure;
+
+import lombok.Getter;
+import org.corfudb.infrastructure.logreplication.LogReplicationConfig;
+
+import java.util.Objects;
+
+import static org.corfudb.infrastructure.logreplication.infrastructure.plugins.DefaultLogReplicationConfigAdapter.SAMPLE_CLIENT;
+
+/**
+ * This class represents a session/connection with a remote cluster.  Each session has a dedicated Replication FSM and
+ * Snapshot and LogEntry Sync Readers/Writers.
+ *
+ * A session is uniquely identified with the following parameters:
+ * 1) Cluster Id of the Remote Cluster
+ * 2) Replication Model being used by the Remote Cluster
+ * 3) Application(Client) consuming the data replicated by this model
+ *
+ * A given remote cluster can have multiple clients and multiple replication models.  Additionally, multiple clients
+ * can use the same replication model.
+ * There will be an instance of this class corresponding to each combination of (remote cluster, replication
+ * model, client).
+ */
+public class CorfuReplicationSession {
+
+    @Getter
+    private final String remoteClusterId;
+
+   @Getter
+   private final CorfuReplicationSubscriber subscriber;
+
+    public CorfuReplicationSession(String remoteClusterId, CorfuReplicationSubscriber subscriber) {
+        this.remoteClusterId = remoteClusterId;
+        this.subscriber = subscriber;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        CorfuReplicationSession that = (CorfuReplicationSession) o;
+        return remoteClusterId.equals(that.remoteClusterId) && subscriber.equals(that.subscriber);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(remoteClusterId, subscriber);
+    }
+
+    // TODO: To be removed after session is introduced in connections
+    public static CorfuReplicationSession getDefaultReplicationSessionForCluster(String remoteClusterId) {
```

Comment:
Just curious as to why this static method was needed ?

Reply:
The main use of it is in LogReplicationServer - handleLrEntryRequest() and handleMetadataRequest().  With this change, we now use ReplicationSession to get the Sink Manager.  However, the incoming connection header does not yet have the information to construct the session.  Till it is available, just using a default session object.

The other use is in forcedSnapshotSync where I have a TODO that we need to define if this operation should be for all models of a cluster or session-specific.  If it is session-specific, we will get the additional params from the request.  Till it is decided, I am using the default (model + client) combination.